doc/source/ray-job-submission/overview.rst

```diff
@@ -0,0 +1,239 @@
+
+.. code-block:: bash
+
+  | your_working_directory ("./")
+  | ├── script.py
+
+.. code-block:: python
+
+    import ray
+    import requests
+
+    ray.init()
+
+    @ray.remote
+    class Counter:
+        def __init__(self):
+            self.counter = 0
+
+        def inc(self):
+            self.counter += 1
+
+        def get_counter(self):
+            return self.counter
+
+    counter = Counter.remote()
+
+    for _ in range(5):
+        ray.get(counter.inc.remote())
+        print(ray.get(counter.get_counter.remote()))
+
+    print(requests.__version__)
+
+Ray Job SDK
+------------
+
+Ray job sdk is the recommended way to submit jobs programmatically.
+
+| Start a local ray cluster headnode
+
+.. code-block:: bash
+
+    ray start --head
+
+We can import and intialize job submission client by providing an valid ray cluster headnode address where port is same as ray dashboard. We're using your local ray cluster as example but it works the same for remote ray cluster addresses.
+
+.. code-block:: python
+
+    from ray.dashboard.modules.job.sdk import JobSubmissionClient
+
+    client = JobSubmissionClient("http://127.0.0.1:8265")
+
+Then we can submit our application to ray cluster via job SDK.
+
+.. code-block:: python
+
+    job_id = client.submit_job(
+        # Entry point to execute
+        entrypoint="python script.py",
+        # Working dir
+        runtime_env={
+            "working_dir": "./",
+            "pip": ["requests==2.26.0"]
+        }
+    )
+
+.. tip::
+
+    By default ray job server will generate a new uuid as return value, but you can also generate your unique job_id first and pass it into :code:`submit_job`. In this case the job will be executed with your given id, and will throw error if same job_id is submitted more than once for the same ray cluster. This can facilitate job id management assuming you have your own job registry to generate and persist unique job ids.
+
+Now we can have a simple polling loop that checks job status until it reaches terminal state, and get logs at the end. We are expected to see actor printed numbers as well as correct version of :code:`requests` module used via runtime_env.
+
+.. code-block:: python
+
+    from ray.dashboard.modules.job.common import JobStatus, JobStatusInfo
+
+    def wait_until_finish(job_id):
+        start = time.time()
+        timeout = 5
+        while time.time() - start <= timeout:
+            status_info = client.get_job_status(job_id)
+            status = status_info.status
+            print(f"status: {status}")
+            if status in {JobStatus.SUCCEEDED, JobStatus.STOPPED, JobStatus.FAILED}:
+                break
+            time.sleep(1)
+
+
+    wait_until_finish(job_id)
+    logs = client.get_job_logs(job_id)
+    print(logs)
+
+.. tip::
+
+    We can also use other remote uris for runtime env, such as S3 or GIT. See "Remote URIs" section of :ref:`Runtime Environments<runtime-environments>` for details.
```

Comment:
It should be possible to link to the exact section, I think one of these should work?
```
:doc:`remote-uris`
:ref:`remote-uris`
```

Reply:
```suggestion
    We can also use other remote uris for runtime env, such as S3 or Git. See "Remote URIs" section of :ref:`Runtime Environments<runtime-environments>` for details.
```